src/v/cloud_io/remote.cc

```diff
@@ -403,6 +410,16 @@ ss::future<download_result> remote::download_stream(
                 uint64_t content_length = co_await cons_str(
                   length, std::move(throttled_st));
                 transfer_details.on_success_size(content_length);
+
+                /// Notify external subscribers about the download activity
+                /// after it completes and we know its size.
+                notify_external_subscribers(
```

Comment:
This should be moved after the try/catch block otherwise won't be notified if throttle_download throws or something in between.

Reply:
The notifications are now post operation. I'm planning to add new notification type that will send information about errors to the subscriber. This PR is only for moving + some minimal changes.